### PR TITLE
fix(styles): Update preferences to be scrollable with new styling

### LINKED ISSUE #3256: New Tab Preferences menu is not scrollable if the browser is resized and the menu is not fully displayed
Mirror of bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1394409

> User Agent: Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0
Build ID: 20170828100127
>
> [Prerequisites]:
> - From about:config make the following changes:
> browser.newtabpage.activity-stream.enabled = true
>
> [Steps]:
> 1. Open Nightly browser and open a new tab.
> 2. Click on the "Customize your New Tab page" setting button.
> 3. Resize the browser window so that the settings are not fully displayed.
> 4. Scroll to see all the settings available.
>
> [Expected results]:
> - When there are more settings than the ones displayed in viewport, a scroll bar is displayed for > "New Tab Preferences" menu.
>
> [Actual results]:
> - Scroll bar for settings menu is not available

### LINKED ISSUE #3273: Move line separators in preferences and differently style confirmation
See https://mozilla.invisionapp.com/share/NYDAMWECB#/251367746_Preferences_Spec

Between search / top sites
image

and between highlights / snippets
image

and background of "Done"
image

and lighter intro text:
image

### PR DESCRIPTION
**Pull request by Mardak**
Fix #3256 and fix #3273. r?@rlr 

short:
image

tall:
image

### CODE REVIEW

File: system-addon/content-src/components/PreferencesPane/PreferencesPane.jsx [hunk 1]
```diff
@@ -86,6 +86,8 @@ class PreferencesPane extends React.Component {
               <PreferencesInput className="showSearch" prefName="showSearch" value={prefs.showSearch} onChange={this.handlePrefChange}
                 titleString={{id: "settings_pane_search_header"}} descString={{id: "settings_pane_search_body"}} />
 
+              <hr />
```
Comment (rlr) [inline system-addon/content-src/components/PreferencesPane/PreferencesPane.jsx]: i forgot these things existed 👀

**Comment by rlr** [comment]
I think you need to add something like:  `.prefs-modal-inner-wrapper { padding-bottom: 80px; }`

Otherwise the button section can end up hiding the very bottom of the form in short windows:

screen shot 2017-08-31 at 6 38 26 pm

File: system-addon/content-src/components/PreferencesPane/_PreferencesPane.scss [hunk 2]
```diff
@@ -52,18 +64,23 @@
...
 
       > label {
-        font-size: 16px;
+        font-size: 17px;
         font-weight: bold;
+        line-height: 19px;
+      }
+
+      &:last-child {
+        margin-bottom: 100px;
```
Comment (Mardak) [inline system-addon/content-src/components/PreferencesPane/_PreferencesPane.scss]: this should prevent the last item from being hidden.

**Comment by Mardak** [comment]
Let's go with the padding on the parent instead of margin on the last child.

**Comment by as-pine-proxy** [comment]
[Treeherder: c54fec76f6f34f9efd75365386d023e9d636ae55](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=c54fec76f6f34f9efd75365386d023e9d636ae55)

**Pull request merged**